contracts/services/forwarder/src/tests.rs

```diff
@@ -98,192 +48,105 @@ impl ForwarderTestSuite {
             crate::contract::query,
         );
 
-        let forwarder_code_id = app.store_code(Box::new(forwarder_code));
+        let forwarder_code_id = inner.app_mut().store_code(Box::new(forwarder_code));
 
         Self {
-            app,
-            owner,
-            processor,
-            account_code_id,
-            account_salt: salt,
+            inner,
             forwarder_code_id,
             input_addr,
             output_addr,
+            input_balances,
         }
     }
 
-    fn api(&self) -> &MockApi {
-        self.app.api()
-    }
-
-    fn instantiate(&mut self, svc_cfg: &ServiceConfig) -> AnyResult<Addr> {
-        let msg = InstantiateMsg {
-            owner: self.owner.to_string(),
-            processor: self.processor.to_string(),
-            config: svc_cfg,
+    pub fn forwarder_init(&mut self, cfg: &ServiceConfig) -> Addr {
+        let init_msg = InstantiateMsg {
```

Comment:
My personal preference is to use the whole type path for contract common types, make its much easier to understand of which contract this type is, without "verifying" this is the correct type.
Less necessary here, because of the function name, but still a bit nicer.

Reply:
I see what you mean, I agree. But probably less of a big deal here as the tests are in the contract's crate.